gulpfile.js

```diff
@@ -333,6 +333,15 @@ export async function migrate() {
     logger: console
   });
 
+  const executedMigrationNames = (await umzug.executed()).map(migration => migration.name);
+
+  migrationsToRun.forEach(migrationFullPath => {
+    const executedMigrationName = executedMigrationNames.find(migrationName => migrationFullPath.includes(migrationName));
+    if (executedMigrationName) {
+      console.log(`Migration ${executedMigrationName} was already run, skipping it.`);
+    }
+  });
+
```

Comment:
BTW. there's also `umzug.up({ rerun: true })` in case you want to give the user the chance to rerun. There's also a function on `umzug` that would tell you which migrations are already run (i.e. in the DB table). That means, we could change the logic to show the user in the list which migrations have already run (and would be rerun if selected) and which are still missing. Then I don't know if it's worth it, just saying. I'll approve anyway.

Reply:
That's cool, that you could rerun a migration. Actually I was also tempted to extract the migration logic in the `dev` folder, but then thought better to first see which direction we actually go with migrations.
Same with your suggestions, I guess, before spending too much time with this we should see how we deal with migrations next.